### PR TITLE
feat(Bonus Pagamenti Digitali): [#175269135,#175271447] BPD Transaction summary component with BottomSheet Info box

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR adds the BPDTransactionSummaryComponent to transactions screen with it's information BottomSheetModal

### CODE REVIEW

File: ts/features/bonus/bpd/components/BPDTransactionSummaryComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      <Body>
+        {I18n.t("bonus.bpd.details.transaction.detail.summary.body.text1")}
+        <H4 weight={"Bold"}>{`${format(
+          props.period.startDate,
+          "DD MMM YYYY"
+        )} - ${format(props.period.endDate, "DD MMM YYYY")} `}</H4>
+        {I18n.t("bonus.bpd.details.transaction.detail.summary.body.text2")}
+        <H4 weight={"Bold"}>
+          {I18n.t(
+            "bonus.bpd.details.transaction.detail.summary.body.text3" as TranslationKeys,
```
Comment (Undermaken) [inline ts/features/bonus/bpd/components/BPDTransactionSummaryComponent.tsx:84]: Perhaps here we could use code merged a few minutes ago ;)

File: ts/features/bonus/bpd/components/BPDTransactionSummaryComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            {
+              count: props.totalAmount.transactionNumber
+            }
+          )}
+        </H4>{" "}
```
Comment (Undermaken) [inline ts/features/bonus/bpd/components/BPDTransactionSummaryComponent.tsx:89]:
```suggestion
        </H4>{` ${I18n.t("bonus.bpd.details.transaction.detail.summary.body.text4")}`}
```

File: ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx [hunk 3]
```diff
@@ -27,29 +32,46 @@ const dataForFlatList = (
...
+  );
+
+  return (
+    <BaseScreenComponent goBack={true} headerTitle={I18n.t("bonus.bpd.title")}>
+      <SafeAreaView style={IOStyles.flex}>
+        <View style={[IOStyles.horizontalContentPadding, IOStyles.flex]}>
+          <View spacer={true} large={true} />
+          <H1>{I18n.t("bonus.bpd.details.transaction.title")}</H1>
+          <View spacer={true} />
+          {pot.isSome(props.selectedAmount) && props.selectedPeriod && (
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx:49]: Using `format(trxSortByDate[0].trxDate` we assume that array will be always not empty.
This is a strong assumption and could generate an error in specific conditions

```typescript
import { index } from "fp-ts/lib/Array";
const maybeFirstDate = index(0,trxSortByDate).map(i => i.trxDate);
...
{pot.isSome(props.selectedAmount) && props.selectedPeriod && maybeFirstDate.isSome()
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/components/BPDTransactionSummaryComponent.tsx (hunk 1 above)
Comment (CrisTofani) [reply ts/features/bonus/bpd/components/BPDTransactionSummaryComponent.tsx:84]: Yes I was actually waiting for it! 😄

File: ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx (hunk 3 above)
Comment (CrisTofani) [reply ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx:49]: You're right, I made an additional fix.
Using this three conditions together (`pot.isSome(props.selectedAmount) && props.selectedPeriod && maybeFirstDate.isSome()`) may cause to not show the summary component even if it should be displayed.

For example: The selected period and amount are correctly available, but no transactions is still available, the condition above would hide all the text block.

I think this solution may cover that case easier:
```javascript
  const lastUpdateDate = format(
    index(0, trxSortByDate).fold(new Date(), t => t.trxDate),
    "DD MMMM YYYY"
  );
```

Comment (Undermaken) [reply ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx:55]: @CrisTofani for me, it's ok.
So the fallback, if the date array is empty, will be `now`.
Do you confirm?

**Review by Undermaken — APPROVED** [review]

File: ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx (hunk 3 above)
Comment (CrisTofani) [reply ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx:55]: yes it is!

**Review by Undermaken — APPROVED** [review]

**Pull request merged**